### PR TITLE
Added subgrid integration for ps_bar, unified unit tests, removed loops from szar.clustering

### PR DESCRIPTION
**Pull request by dylancromer**
Pretty much what is in the title.

I found a _mostly_ vectorized way to do the subgrid integration which is far faster when composing the fine_(blah) functions (it got pretty slow when it came time to do subgrid integration for ps_bar - about 40 seconds! So this addresses that).

The unit tests are also now in a single script so they can all be run at once, without having to run the class init scripts more than once. They run quite quickly and generate the same figures we have been using for our core tests.

### CODE REVIEW

File: szar/clustering.py [hunk 1]
```diff
@@ -130,78 +130,105 @@ def fine_sfunc(self, fsky, nsubsamples):
...
+        dvdz = np.array([self.dVdz_fine(zs) for zs in fine_zgrid])
+
+        dz = fine_zgrid[0,1] - fine_zgrid[0,0]
+
+        assert np.allclose(dz * np.ones(tuple(np.subtract(fine_zgrid.shape, (0,1)))),  np.diff(fine_zgrid,axis=1), rtol=1e-3)
+
+        integral = simps(dvdz * ntils**2, dx=dz)
+        integral *= 4 * np.pi * fsky
+
+        return integral
```
Comment (nbatta) [inline szar/clustering.py]: Did you test this new volume integral?

File: szar/clustering.py [hunk 2]
```diff
@@ -130,78 +130,105 @@ def fine_sfunc(self, fsky, nsubsamples):
...
+        dz = fine_zgrid[0,1] - fine_zgrid[0,0]
 
-    def ps_tilde(self,mu):
+        assert np.allclose(dz * np.ones(tuple(np.subtract(fine_zgrid.shape, (0,1)))),  np.diff(fine_zgrid,axis=1), rtol=1e-3)
         
-        beff_arr = np.outer(np.ones(len(mu)), self.b_eff_z())
-        mu_arr = np.outer(mu, np.ones(len(self.b_eff_z())))
-        logGrowth = np.outer(np.ones(len(mu)), self.cc.fgrowth(self.HMF.zarr))
-        prefac = (beff_arr + logGrowth*mu_arr**2)**2
+        integral = simps(dvdz, dx=dz)
```
Comment (nbatta) [inline szar/clustering.py]: Can you do this without calling simps?

File: szar/clustering.py (hunk 1 above)
Comment (dylancromer) [reply szar/clustering.py]: Yes, I re-ran the tests for V_eff and V_0 and the changes pass (and in fact make our plots of V_eff closer to that of the paper we were comparing to - see Nuclino for this).

File: szar/clustering.py (hunk 2 above)
Comment (dylancromer) [reply szar/clustering.py]: Trapz would work too, but I have tested the timing for trapz vs simps and it is not very noticeable. The most time trapz ever bought over simps was about 5s, when calling what was essentially a nested loop which performed integrals each iteration. Without loops we're talking sub-second differences (and the loops are gone).

**Comment by dylancromer** [comment]
Pushed one fix for szar.clustering.V_eff.

**Comment by nbatta** [comment]
looks good to me

**Pull request merged**